connector_importer/models/sources/source_consumer_mixin.py

```diff
@@ -0,0 +1,72 @@
+# Author: Simone Orsi
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, tools
+
+
+class ImportSourceConsumerMixin(models.AbstractModel):
+    """Source consumer mixin.
+
+    Inheriting models can setup, configure and use import sources.
+
+    Relation towards source records is generic to grant maximum freedom
+    on which source type to use.
+    """
+
+    _name = "import.source.consumer.mixin"
+    _description = "Import source consumer"
+
+    source_id = fields.Integer(string="Source ID", required=False, ondelete="cascade")
+    source_model = fields.Selection(
+        string="Source type", selection="_selection_source_ref_id"
+    )
+    source_ref_id = fields.Reference(
+        string="Source",
+        compute="_compute_source_ref_id",
+        selection="_selection_source_ref_id",
+        # NOTE: do not store a computed fields.Reference, Odoo crashes
+        # with an error message "Mixing appels and orange..." when performing
+        # a self.recompute() on such fields.
+        store=False,
+    )
+    source_config_summary = fields.Html(
+        compute="_compute_source_config_summary", readonly=True
+    )
+
+    @api.depends("source_model", "source_id")
+    def _compute_source_ref_id(self):
+        for item in self:
+            item.source_ref_id = False
+            if not item.source_id or not item.source_model:
+                continue
+            item.source_ref_id = "{0.source_model},{0.source_id}".format(item)
+
+    @api.model
+    @tools.ormcache("self")
```

Comment:
???'@tools.ormcache("self")`????

Reply:
it's equivalent to not pass anything. BUT I think it's just a left over from older implementations as there's nothing to compute and cache here... I'll drop it. Thanks for having a look ;)